main.ml

```diff
@@ -194,13 +196,27 @@ end = struct
 
   let sr_create device_config =
     compat_uri device_config >>>= fun compat_in ->
-    let compat_out rpc =
-      (* The PVS version will return nothing *)
-      if rpc = R.Null then
-        Rpcmarshal.marshal Xapi_storage.Control.typ_of_configuration device_config
-      else rpc
+    let compat_out =
+      if !V.version = Some pvs_version then begin
+        fun rpc ->
+          (* The PVS version will return nothing *)
+          if rpc = R.Null then
+            Rpcmarshal.marshal Xapi_storage.Control.typ_of_configuration device_config
+          else rpc
+      end
+      else id
     in
     return (Ok (device_config, compat_in, compat_out))
+
+  let sr_attach ~uuid device_config =
+    compat_uri device_config >>>= fun compat_in ->
+    let device_config =
+      if !V.version = Some pvs_version then
+        ("sr_uuid", uuid) :: device_config
```

Comment:
PVS did not use `sr_uuid`, that was something that older versions of our own xapi-storage-plugins used.
I think you can safely drop `sr_uuid` from here, and keep just `compat_uri`